website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,447 @@
+![An example of a Mipmap texture](../../../images/opengl/chapter1/lesson3/example_mipmap.png)
+
+But what is this used for?
+
+At first look, mipmaps may seem very useless. But in reality, they are extremely important! Mipmaps are used when the texture is being rendered at a very low size,
+which can cause some weird visual artifacts, even with the use of the linear filter. The most potent visual effect is something called the "moiré effect", which can
+be tiring for the eyes, and generally look quite ugly.
+
+The purpose of a mipmap is to provide an alternative texture to the fragment shader, a one with a more appropriate size for the geometry being drawn, almost completely
+eliminating the moiré effect.
+
+An example from [Wikipedia](https://en.wikipedia.org/wiki/File:Mipmap_Aliasing_Comparison.png). It's possible to notice weird patterns
+generated far away in the render without mipmaps:
+
+![Mipmap usage example](../../../images/opengl/chapter1/lesson3/mipmap_comparation.png)
+
+But if you think that generating mipmaps by hand for all your textures is really hard work, don't worry! OpenGL provides a special method to
+do this for you.
+
+Just after setting the texture parameters, try to add this line to your code:
+```c#
+_gl.GenerateMipmap(TextureTarget.Texture2D);
+```
+
+But this is not everything, we also need to set some parameters to tell the sampler to use the mipmaps:
+
+### `GL_NEAREST_MIPMAP_NEAREST`, `GL_LINEAR_MIPMAP_NEAREST`, `GL_NEAREST_MIPMAP_LINEAR` & `GL_LINEAR_MIPMAP_LINEAR`:
+These parameters should be used as the Texture Min filter and only with textures that have a mipmap.
+
+#### `TextureMinFilter.GL_NEAREST_MIPMAP_NEAREST`:
+Will get the mip with the closest size of the final image and will return the color of the pixel that matches exactly with the texture coordinates.
+
+#### `TextureMinFilter.GL_LINEAR_MIPMAP_NEAREST`:
+Will get the color of the pixels that match exactly with the texture coordinates of each mipmap and interpolates them.
+
+#### `TextureMinFilter.GL_NEAREST_MIPMAP_LINEAR`:
+Will get the mip with the closest size of the final image and will return the interpolated color
+of the pixels surrounding the pixel in the texture coordinate.
+
+#### `TextureMinFilter.GL_LINEAR_MIPMAP_LINEAR`:
+Will get the interpolated color of the pixels surrounding the pixel that matches exactly with the texture coordinates of each mipmap and interpolates them.
+
+Knowing this, you can change the parameters for:
+```c#
+_gl.TextureParameter(_texture, TextureParameterName.TextureWrapS, (int)TextureWrapMode.Repeat);
+_gl.TextureParameter(_texture, TextureParameterName.TextureWrapT, (int)TextureWrapMode.Repeat);
+_gl.TextureParameter(_texture, TextureParameterName.TextureMinFilter, (int)TextureMinFilter.NearestMipmapNearest); // <- change here!
+_gl.TextureParameter(_texture, TextureParameterName.TextureMagFilter, (int)TextureMagFilter.Nearest);
```

Comment:
There are no words about this line/magnification filters.

Reply:
Paragraph suggestion: "When we draw a texture, the area on the screen in which we're drawing the texture typically doesn't have the same size or shape as the texture. This means that during rendering we usually have to either scale up or down the size of the texture! This process is called _filtering_.

Reply:
i tried to fix this and the request above at the same time, pls rate it in the next commit

Reply:
I'm closing conversations on filtering and unifying/superseding them with [this one](https://github.com/dotnet/Silk.NET/pull/1899/files#r1477123745).